### ISSUE TITLE
Bug: When attempting to format JavaScript outside of project, the extension is reported as "not available."

### ISSUE THREAD

**Issue by DanKaplanSES** [eslint/eslint#17800]
### Environment

Environment Info:

Node version: v18.17.1
npm version: v9.6.7
Local ESLint version: v8.50.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "jasmine": true
  },
  "plugins": [],
  "extends": [
    "eslint:recommended",
    "prettier",
    "plugin:prettier/recommended"
  ],
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly",
    "State": "writable",
    "setup": "writable"
  },
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "prettier/prettier": "error",
    "quotes": ["off"],
    "no-plusplus": ["off"],
    "curly": ["error", "all"],
    "prefer-const": ["warn"]
  }
}

```
</details>

1. Save this file outside of your project:
    ```js
    // test.js
    const x = function () {
    console.log(
    `log!`)
    };
    ```
1. Press <kbd>ctrl</kbd> + <kbd>o</kbd> to open the file browser
2. Open the file you saved on step one
3. Press <kbd>ctrl</kbd> + <kbd>shift</kbd> + <kbd>p</kbd> to open the command palette
4. Select "Format Document"

### What did you expect to happen?

I expected this JavaScript file to be formatted by ESLint. 

If it doesn't work for some reason, I expected to see a helpful log message in the output tab.

### What actually happened?

I get this error message:

![image](https://github.com/eslint/eslint/assets/373675/75fd40f3-3f51-4d76-84aa-9d3e815d104d)

Pressing that configure button _does not_ let me select `eslint` as an option.

Nothing is logged to the eslint output tab. I have debugging on and I have server trace on. 

### Link to Minimal Reproducible Example

https://github.com/DanKaplanSES/jsdom-sandbox

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

You will still have to follow the steps in the minimal reproduction example: I can't commit a file outside of its repository and that is one of the steps.

**Comment by Rec0iL99** [comment]
Hi @DanKaplanSES, thanks for the issue. I think this is happening because for the `test.js` file there is no ESLint config file relative to it or in the parent directories above it. 

```bash
root/
    test.js
    project/
        index.js
        .eslintrc.json
```

The error is coming from [`vscode-eslint`](https://github.com/Microsoft/vscode-eslint) which is an extension that is not maintained by the ESLint team. Please consult the maintainers there for more help.

**Comment by DanKaplanSES** [comment]
> The error is coming from [`vscode-eslint`](https://github.com/Microsoft/vscode-eslint) which is an extension that is not maintained by the ESLint team. Please consult the maintainers there for more help.

@Rec0iL99  🤦‍♂️ That's where I _meant_ to open this issue. Sorry for not paying closer attention. I'll re-create that there, as I originally intended
